apps/frontend/src/libs/components/checkbox/checkbox.tsx

```diff
@@ -1,73 +1,141 @@
+import { ZERO_INDEX } from "~/libs/constants/constants.js";
+import { getValidClassNames } from "~/libs/helpers/helpers.js";
 import { useCallback, useFormController } from "~/libs/hooks/hooks.js";
 import {
-	type Control,
-	type FieldPath,
 	type FieldValues,
+	type FormFieldProperties,
 	type InputOption,
 } from "~/libs/types/types.js";
 
 import styles from "./styles.module.css";
 
 type Properties<T extends FieldValues> = {
-	control: Control<T, null>;
+	hasVisuallyHiddenLabel?: boolean;
 	label: string;
-	name: FieldPath<T>;
-	options: InputOption[];
-};
+	layout?: "column" | "wrap";
+	onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+	options: InputOption<number>[];
```

Comment:
Checkbox value can be a string as well. Maybe add another generic to a component and use it